src/modules/cpu/kernel/flip_voxel.hpp

```diff
@@ -0,0 +1,420 @@
+        roiDefault = {0, 0, 0, (Rpp32s)srcGenericDescPtr->dims[3], (Rpp32s)srcGenericDescPtr->dims[2], (Rpp32s)srcGenericDescPtr->dims[1]};
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
+    {
+        RpptROI3D roi;
+        RpptROI3DPtr roiPtrInput = &roiGenericPtrSrc[batchCount];
+        compute_roi3D_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32u horizontalFlag = horizontalTensor[batchCount];
+        Rpp32u verticalFlag = verticalTensor[batchCount];
+        Rpp32u depthFlag = depthTensor[batchCount];
+
+        Rpp32f *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrImage = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        Rpp32u bufferLength = roi.xyzwhdROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u vectorIncrement = 24;
+        Rpp32u vectorIncrementPerChannel = 8;
+        Rpp32u alignedLength = (bufferLength / vectorIncrement) * vectorIncrement;
+
+        // Initialize load functions with default values
+        auto load24FnPkdPln = &rpp_load24_f32pkd3_to_f32pln3_avx;
+        auto load8Fn = &rpp_load8_f32_to_f32_avx;
+
+        // Update the load functions, hFactor, vFactor and hStrideSrcIncrement based on the flags enabled
+        if (horizontalFlag == 1)
+        {
+            load24FnPkdPln = &rpp_load24_f32pkd3_to_f32pln3_mirror_avx;
+            load8Fn = &rpp_load8_f32_to_f32_mirror_avx;
+        }
+
+        Rpp32f *srcPtrChannel, *dstPtrChannel;
+        dstPtrChannel = dstPtrImage;
+
+        // Compute constant increment, Decrement factors used in source pointer updation
+        Rpp32s srcPtrIncrement = (horizontalFlag)? -vectorIncrement : vectorIncrement;
+        Rpp32u hFlipFactor = (vectorIncrement - layoutParams.bufferMultiplier) * horizontalFlag;
+        Rpp32s srcPtrIncrementPerChannel = (horizontalFlag)? -vectorIncrementPerChannel : vectorIncrementPerChannel;
+        Rpp32u hFlipFactorPerChannel = (vectorIncrementPerChannel - 1) * horizontalFlag;
+        Rpp32s srcPtrIncrementPerRGB = (horizontalFlag) ? -3 : 3;
+        Rpp32s srcPtrIncrementPerPixel = (horizontalFlag) ? -1 : 1;
+
+        // flip without fused output-layout toggle (NCDHW -> NCDHW)
+        if ((srcGenericDescPtr->layout == RpptLayout::NCDHW) && (dstGenericDescPtr->layout == RpptLayout::NCDHW))
+        {
+            alignedLength = (bufferLength / 8) * 8;
+            Rpp32u hFactor = roi.xyzwhdROI.xyz.x * layoutParams.bufferMultiplier;
+            Rpp32u vFactor = roi.xyzwhdROI.xyz.y * srcGenericDescPtr->strides[3];
+            Rpp32u dFactor = roi.xyzwhdROI.xyz.z * srcGenericDescPtr->strides[2];
+            Rpp32s hStrideSrcIncrement = srcGenericDescPtr->strides[3];
+            Rpp32s depthStrideIncrement = srcGenericDescPtr->strides[2];
```

Comment:
Consistency on variable names for readability - Pick either say yStrideSrcIncrement & zStrideSrcIncrement OR heightStrideSrcIncrement & depthStrideSrcIncrement?